types/src/account_config/resources/diem_id.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{account_address::AccountAddress, event::EventHandle, on_chain_config::OnChainConfig};
+use move_core_types::move_resource::MoveResource;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct DiemIdDomains {
+    domains: Vec<DiemIdDomain>,
+}
+
+impl DiemIdDomains {
+    pub fn domains(&self) -> &[DiemIdDomain] {
+        &self.domains
+    }
+}
+
+impl MoveResource for DiemIdDomains {
+    const MODULE_NAME: &'static str = "DiemId";
+    const STRUCT_NAME: &'static str = "DiemIdDomains";
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct DiemIdDomain {
+    domain: String,
+}
```

Comment:
Is there a reason you're creating a new type here instead of using the one you're creating in this PR? https://github.com/diem/diem/pull/8362

Reply:
I will be, I was just unblocking myself because i don't have the commits stacked